### PR TITLE
refs #3079: Make Counter::inc() return count to align with dec()

### PR DESCRIPTION
**Pull request by geckon**

### CODE REVIEW

File: include/qore/QoreCounter.h [hunk 1]
```diff
@@ -59,7 +59,11 @@ class QoreCounter {
    DLLEXPORT void destructor(ExceptionSink* xsink);
 
    //! increments the counter
-   DLLEXPORT void inc();
+   /** @return the current value after the increment
+
+       @since %Qore 0.8.13 the current value is returned
```
Comment (geckon) [inline include/qore/QoreCounter.h]: Will this be a part of Qore 0.8.13.9 or should this say 0.8.14 or maybe something else? Thanks.

File: lib/QC_Counter.qpp [hunk 2]
```diff
@@ -89,9 +89,13 @@ Counter::copy() {
     @code{.py}
 counter.inc();
     @endcode
+
+    @return the current value after the increment is returned
+
+    @since %Qore 0.8.13 this method returns the current value after the increment
```
Comment (geckon) [inline lib/QC_Counter.qpp]: Will this be a part of Qore 0.8.13.9 or should this say 0.8.14 or maybe something else? Thanks.

File: include/qore/QoreCounter.h (hunk 1 above)
Comment (davidnich) [reply include/qore/QoreCounter.h]: this is target to develop, so the version will be 0.9 in fact

File: lib/QC_Counter.qpp (hunk 2 above)
Comment (davidnich) [reply lib/QC_Counter.qpp]: develop = 0.9

**Review by davidnich — CHANGES_REQUESTED** [review]
this is fine, just please add a release note in `doxygen/lang/900_release_notes.dox.tmpl` under "new and updated methods in existing classes"

File: include/qore/QoreCounter.h (hunk 1 above)
Comment (geckon) [reply include/qore/QoreCounter.h]: Thanks

**Comment by geckon** [comment]
Updated

**Review by davidnich — APPROVED** [review]

**Pull request merged**